node/src/components/small_network.rs

```diff
@@ -811,12 +875,23 @@ where
         ret
     }
 
-    /// Returns whether or not this node has been isolated.
-    ///
-    /// An isolated node has no chance of recovering a connection to the network and is not
-    /// connected to any peer.
-    fn is_isolated(&self) -> bool {
-        self.pending.is_empty() && self.outgoing.is_empty() && self.incoming.is_empty()
+    /// Returns whether or not this node has been disconnected from all known nodes.
+    fn is_not_connected_to_any_known_address(&self) -> bool {
+        for &known_address in &self.known_addresses {
+            if self.pending.contains_key(&known_address) {
+                return false;
+            }
+
+            if self
+                .outgoing
+                .values()
+                .any(|outgoing_connection| outgoing_connection.peer_address == known_address)
+            {
+                return false;
```

Comment:
Why is the asymetric connection considered as "connected"?

Reply:
This might be a naming problem; this function used to be called `is_isolated`. Our conditions are that if we lost the outgoing connection to all known peers we are what we formerly called "isolated" and attempt to remedy that.

Incoming connections are ignored to avoid issues where a stray incoming connection prevents us from triggering reconnection.

Reply:
Thanks